app/helpers/brevo.py

```diff
@@ -178,17 +193,39 @@ def update_deal_stage(self, data: UpdateDealStageData):
             raise BrevoRequestError(f"Request to Brevo API failed: {e}")
 
     @check_api_key
-    def get_deals_by_pipeline(self, data: GetDealsByPipelineData):
+    def get_all_deals_by_pipeline(self, data: GetAllDealsByPipelineData):
         try:
-            url = f"{self.BASE_URL}crm/deals"
-            params = {
-                "filters[attributes.pipeline]": data.pipeline_id,
-                "limit": data.limit,
-            }
-            response = self._session.get(url, params=params)
-            response.raise_for_status()
-            return response.json()
-        except ApiException as e:
+            all_deals = []
+            offset = 0
+            limit = 100  # Brevo's maximum limit per request (https://developers.brevo.com/docs/api-limits#general-rate-limiting)
+
+            while True:
+                params = {
+                    "filters[attributes.pipeline]": data.pipeline_id,
+                    "offset": offset,
+                    "limit": limit,
+                }
+
+                response = self._session.get(
+                    f"{self.BASE_URL}crm/deals", params=params
+                )
+                # Retry after delay if rate-limited (https://developers.brevo.com/docs/api-limits#rate-limit-reset)
+                if response.status_code == 429:
+                    time.sleep(
```

Comment:
Attention au `time.sleep` car il laisse le thread en cours même s'il ne fait rien. 
Comment est exécuté ce code ? Dans un scheduler ? 
- Si oui n'y a-t-il pas de risque qu'on dépasse les 15mn ?
- Si non, on monopolise un des 2 workers du container back

Je ne suis pas fan de la manière dont tu gères le rate limit : 
- vérifier le status code et arrêter le process si 429 est une bonne chose je pense
- mais il faudrait faire en sorte par ailleurs que cela ne se produise pas : c'est d'ailleurs conseillé dans leur [doc](https://developers.brevo.com/docs/api-limits#how-to-prevent-my-api-requests-from-being-rejected-) : 
> Make sure you properly calculate your rate limit allowance. You need to send your bulk requests equally distributed through the specified time granularity. E.g 1000 RPS = up to 60K requests per minute.

On peut par exemple imaginer si c'est un job qu'il soit lancé plus souvent, par exemple toutes les 5mn, avec une limite d'appels. En enregistrant quelque part où on en est (comme un cursor), si le cursor est à la fin on arrête direct le process (ça pourrait être la condition du `while`)

Reply:
Bon en fait je ne pense pas que le sujet soit d'actualité vu que la limite d'item reçus peut être bien supérieur à 100, je l'ai mis à 10 000 et je reçois bien les 1225 qui sont renseigné dans Brevo, c'est par défaut lorsqu'on renseigne pas de limite qu'on en reçoit que 100